src/analysis/retail/paladin/holy/modules/beacons/BeaconUptime.tsx

```diff
@@ -379,14 +379,14 @@ class BeaconUptime extends Analyzer {
   }
 
   _endOfFight(event: FightEndEvent) {
-    if (this.hasBoL) {
+    if (this.hasBoF) {
       this._handleBoLEvents(event);
```

Comment:
Should this still be handling BoL events?

Reply:
If you have Beacon of Faith, you also have Beacon of Light and should be handling both buff checks as I understand it from running through the code. Beacon of Virtue was weird but wasn't included in the initial report.

Reply:
Alternatively - we could just not return from the two functions and they'd just fallthrough check each other - but this works too I guess

Reply:
these `has*` look like they're meant to be mutually exclusive (although you technically also use beacon of light when you use beacon of faith). Since the bodies are so simple i'd probably go with something like if/elif/elif/else error to make that assumption clearer in code

Reply:
or refactor the separate `has*` into one `this.beaconStyle` or something with an enum

Reply:
I agree that it should be refactored to be a bit more clear, the module itself is a bit arcane. I didn't want to go to that level of refactor when just trying to get the Beacon of Faith uptime to report more accurately. I'm also perfectly fine closing this PR in favor of another PR being made for the refactor, but I'd prefer to not do that myself as I don't play Holy Paladin.

Reply:
this PR certainly isn't doing any harm so might as well land it and wait for a refactor in a slightly less broken state :)